packages/link/src/Link.ts

```diff
@@ -132,6 +152,202 @@ function eventsListener(
   }
 }
 
+async function handleWalletBrowserEvent(
+  event: MessageEvent<WalletBrowserEventType>
+) {
+  switch (event.data.type) {
+    case 'walletBrowserInjectedWalletSelected': {
+      const payload = event.data.payload
+      try {
+        const result = await connectToSpecificWallet(payload.integrationName)
+        sendMessageToIframe({
+          type: 'SDKinjectedConnected',
+          payload: {
+            accounts: result.accounts,
+            chainId: result.chainId,
+            signedTxHash: result.txSigned
+          }
+        })
+      } catch (error) {
+        sendMessageToIframe({
+          type: 'SDKinjectedDisconnected',
```

Comment:
Can we call it something else? No disconnection happens here. Maybe `SDKinjectedConnectionError`